tavern/internal/c2/api_download_file.go

```diff
@@ -0,0 +1,64 @@
+package c2
+
+import (
+	"bytes"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"realm.pub/tavern/internal/c2/c2pb"
+	"realm.pub/tavern/internal/ent"
+	"realm.pub/tavern/internal/ent/file"
+)
+
+func (srv *Server) DownloadFile(req *c2pb.DownloadFileRequest, stream c2pb.C2_DownloadFileServer) error {
+	ctx := stream.Context()
+
+	// Load File
+	name := req.GetName()
+	f, err := srv.graph.File.Query().
+		Where(file.Name(name)).
+		Only(ctx)
+	if ent.IsNotFound(err) {
+		return status.Errorf(codes.NotFound, "%v", err)
+	}
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to query file (%q): %v", name, err)
+	}
+
+	// Set Header Metadata
+	stream.SetHeader(metadata.Pairs(
+		"sha3-checksum", f.Hash,
```

Comment:
Is this sha3? Our current rust crates I don't think support that. Could you do sha256 or sha1?

Reply:
No, we only support SHA3. We should update our rust crates.